### DISCUSSION TITLE
I would like to modify the directory structure or the file name to perform "refactor" through the language server. I use "Eclipse JDT Language Server", but I don't know where to start to realize this function. Is there any relevant information?

### DISCUSSION THREAD

**Discussion by NightMingGit** [TypeFox/monaco-languageclient#489] [Q&A]
I would like to modify the directory structure or the file name to perform "refactor" through the language server. I use "Eclipse JDT Language Server", but I don't know where to start to realize this function. Is there any relevant information?

![image](https://github.com/TypeFox/monaco-languageclient/assets/92633021/5a3c4de1-3744-47ee-a909-94dcd456b7ed)

**Comment by NightMingGit** [comment]
![image](https://github.com/TypeFox/monaco-languageclient/assets/92633021/8fbac88c-e888-449c-96a1-48beb0976725)

**Comment by NightMingGit** [comment]
i don't know where to start

**Reply by kaisalmen** [reply]
The best source of inspiration is again to be found here:
https://github.com/redhat-developer/vscode-java

**Reply by NightMingGit** [reply]
Is there something simpler, this code is too complicated for me

**Reply by NightMingGit** [reply]
Do you need to use monaco-languageclient to achieve it?

**Reply by kaisalmen** [reply]
JDT support refactoring:
https://github.com/eclipse/eclipse.jdt.ls#features

but you need to support this on the client side, you could start to look here:
https://github.com/redhat-developer/vscode-java/blob/master/src/codeActionProvider.ts

https://github.com/redhat-developer/vscode-java/blob/bd03b35664fd9c72a4de06f1d85ad14302ccca12/src/standardLanguageClient.ts#L547

**Reply by NightMingGit** [reply]
Do I need to manually send commands to the lsp server and then manually accept the commands returned by the server
